pkg/lib/oauth/grant_offline_service.go

```diff
@@ -266,10 +259,11 @@ func (s *OfflineGrantService) housekeepOfflineGrant(ctx context.Context, grant *
 			continue
 		}
 
-		expiry := s.computeRefreshTokenExpiryWithClient(expirableRefreshToken{
-			CreatedAt:    token.CreatedAt,
-			LastAccessAt: lastAccess,
-		}, clientConfig)
+		// Assume other refresh tokens are pre-authentication URL refresh token.
+		// These refresh tokens are never returned.
+		// So their expiry should match the expiry of the access token.
+
+		expiry := token.CreatedAt.Add(clientConfig.AccessTokenLifetime.Duration())
```

Comment:
Not all refresh tokens can be cleaned up. There are two refresh tokens generated, only the one generated with `PreAuthenticatedURLTokenService.ExchangeForAccessToken` can be cleaned up using the access token lifetime. The others are generated using normal token exchange flow and the webapp will use that refresh token.